### DISCUSSION TITLE
Modular data model

### DISCUSSION THREAD

**Discussion by jbemmel** [ipspace/netlab#1339] [General]
In theory, modules could dynamically extend the data model.

For example: A 'lag' module could add a 'lag' link type by extending the 'valid_values' at https://github.com/ipspace/netlab/blob/1f59a603809c10b5c5042f3e607c1f0a96fe7575/netsim/defaults/attributes.yml#L52
instead of modifying that file directly.

What do people think about this? Would it be "too much", over-complicating readability?

**Comment by ipspace** [comment]
Adding a new link type value during the module initialization phase might be better than modifying the attribute definition -- it would prevent someone from using the new link type without the lag module.
